### PR TITLE
[iOS] Fabric: Fixes LayoutConformanceView not work

### PR DESCRIPTION
**Pull request by zhongwuzw**
## Summary:

Fixes iOS LayoutConformanceView not work, seems caused by #48188. @NickGerleman can you help to review please? 

## Changelog:

[IOS] [FIXED] - Fabric: Fixes LayoutConformanceView not work

## Test Plan:

Before:
image

After:
image

### CODE REVIEW

File: packages/react-native/ReactCommon/react/renderer/components/view/LayoutConformanceShadowNode.h [hunk 1]
```diff
@@ -16,9 +16,13 @@ namespace facebook::react {
...
-    YogaLayoutableShadowNode,
-    LayoutConformanceProps>;
+class LayoutConformanceShadowNode final
+    : public ConcreteShadowNode<
+          LayoutConformanceShadowNodeComponentName,
+          YogaLayoutableShadowNode,
+          LayoutConformanceProps> {
+ public:
+  using ConcreteShadowNode::ConcreteShadowNode;
+};
```
Comment (javache) [inline packages/react-native/ReactCommon/react/renderer/components/view/LayoutConformanceShadowNode.h:26]: I don't think this part is required

Comment (zhongwuzw) [reply packages/react-native/ReactCommon/react/renderer/components/view/LayoutConformanceShadowNode.h:26]: @javache Seems this check cannot work if we don't subclass ConcreteShadowNode:

https://github.com/facebook/react-native/blob/06751aa0d1750a66f9d6a0a13e9b61e30d9321bb/packages/react-native/ReactCommon/react/renderer/components/view/YogaLayoutableShadowNode.cpp#L505-L506

Comment (javache) [reply packages/react-native/ReactCommon/react/renderer/components/view/LayoutConformanceShadowNode.h:26]: Oh yeah, interesting. Maybe a different behaviour in compilers.

**Review by javache — APPROVED** [review]

**Comment by NickGerleman** [comment]
Huge thanks! This seemed to work for internal Buck build of RNTester for iOS, but I think it may do some things behind the scenes I didn't realize.

**Pull request closed without merging**